introducing-pgzx.mdx

```diff
@@ -0,0 +1,208 @@
+
+```
+var errctx = pgzx.err.Context.init();
+defer errctx.deinit();
+if (errctx.pg_try()) {
+    // zig code that calls several Postgres C functions.
+} else {
+    return errctx.errorValue();
+}
+```
+
+### Development Environment
+
+pgzx comes with a [Nix flakes](https://nixos.wiki/wiki/Flakes) based development environment for developing extensions as well as pgzx itself. It also comes with a project template which you can use to set up this environment in a new repository. To use it, [install Nix](https://github.com/DeterminateSystems/nix-installer) and then run:
+
+```bash
+mkdir my_extension
+cd my_extension
+nix flake init -t github:xataio/pgzx
+```
+
+Then load the nix shell with:
+
+```bash
+nix develop
+```
+
+The dev environment includes commands to relocate the Postgres source code, to start the server, and so on. The template also comes with a minimal extension and a `build.zig` file with a few common tasks. See the template [README](https://github.com/xataio/pgzx/tree/main/nix/templates/init) for how to build the extension form this point.
+
+### Unit tests
+
+Postgres extensions are typically tested via a tool called `pg_regress`. This is supported by pgzx as well.
+
+But we also wanted to have unit tests. This is a bit tricky because the tests need to be compiled and run in the context of a Postgres instance. Otherwise, they won't be able to interact with the Postgres APIs.
+
+In order to solve this, pgzx registers a custom `run_tests` function via the Function manager. This function can be called from SQL (`SELECT run_tests();`) and it will run the unit tests. A test suite is a Zig struct for which each function whose name starts with test is a unit test. To register a test suite, you would typically do something like this:
+
+```
+comptime {
+    pgzx.testing.registerTests(@import("build_options").testfn, .{Tests});
+}
+```
+
+The [registerTests](https://github.com/xataio/pgzx/blob/9825dde752ee4ace2cbc594332eba100292aafd5/src/pgzx/testing.zig#L39) function is another example of `comptime` usage. It iterates over all the fields of a struct and generates the call to run the tests where the `run_tests()` function is called over SQL.
+
+## Why did we start pgzx
+
+At Xata, we have been working on a new Postgres project, which doesn't really have a name yet, so let's call it "Xata's take at distributed Postgres". It is only in its very early stages, however we'll be open sourcing it soon and we're planning to build it in public. It will be somewhat similar to Citus, yet with some different choices, especially around usage and DX, based on what we learned running Xata for the past couple of years.
+
+For this project, we considered two potential implementation directions:
+
+1.  as an external proxy, like Vitess does for MySQL
+2.  as a Postgres extension, like Citus
+3.  as a fork of Postgres, like Greenplum
```

Comment:
@urso Is this correct?

Reply:
Yeah, I would call Greenplum/Postgres-XL a fork.